lib/src/Pos/Binary/Communication.hs

```diff
@@ -66,6 +69,14 @@ instance Bi MsgBlock where
             1 -> MsgNoBlock <$> decode
             t -> cborError $ "MsgBlock wrong tag: " <> show t
 
+-- Serialize `MsgSerializedBlock` with the property
+-- ```
+-- serialize (MsgBlock b) = serializeMsgSerializedBlock (MsgSerializedBlock $ serialize b)
+-- ```
+serializeMsgSerializedBlock :: MsgSerializedBlock -> BS.ByteString
+serializeMsgSerializedBlock (MsgSerializedBlock b) = "\x82\x0" <> unSerialized b
+serializeMsgSerializedBlock (MsgNoSerializedBlock t) = serialize' (MsgNoBlock t)
```

Comment:
Let's use a lazy bytestring. The `<>` in the first alternative won't cause a copy of the whole block. `serialize (MsgNoBlock t)` will make the second one well-typed.